seed/seedwriter/fetcher.go

```diff
@@ -0,0 +1,89 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package seedwriter
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/asserts"
+)
+
+// SeedAssertionFetcher is a Fetcher which is designed to help with the fetching
+// of assertions during seeding. It keeps track of assertions fetched, and allows
+// for retrieving them at any point in time during seeding. It wraps around the
+// asserts.{SequenceFormingFetcher,Fetcher} interfaces to allow for flexible
+// retrieval of assertions.
+// Optimally we would gather all retrieval logic here related to the fetching of
+// assertions during seeding.
```

Comment:
This sentence should likely also be removed as the TODO was

Reply:
Nice catch, thanks!